packages/@glimmer/debug/lib/vm/snapshot.ts

```diff
@@ -172,6 +173,7 @@ export class VmSnapshotArrayDiff<N extends string | number, T extends unknown[]>
         post = value(after, { ref: `${i}:push` });
       } else if (op === 'retain') {
         post = value(after, { ref: `${i}:after` });
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- exhaustion check
```

Comment:
There's possibly a better way to do this, but the issue is: the linter knows that there's only one
kind of `op` left (`'pop'`) so  it thinks that the check is unnecessary (and could be handled by a
simple `else`). However, it's here to ensure that if `op` gets a new member, the type will fail and
force us to add a new clause.

There are a few places like this, and it would be good to consolidate on a single good solution, but
I think ad-hoc solutions are fine for now, since there aren't a ton of these cases, and the "no
unnecessary check" rule is quite valuable and caught a lot of places where the extra check was
either a mistake or caused confusion.